hibernate-core/src/main/java/org/hibernate/mapping/Index.java

```diff
@@ -191,6 +191,10 @@ public int getColumnSpan() {
 		return columns.iterator();
 	}
 
+	public java.util.Map<Column, String> getColumnOrderMap() {
+		return columnOrderMap;
```

Comment:
I see anyway if I have to choose my vote is for unmodifiableMap :)
naturally if you have nothing against it
 Thanks